unit_tests/UnitTestNgpMesh1.C

```diff
@@ -67,6 +67,84 @@ TEST_F(Hex8MeshWithNSOFields, ngpMesh1)
 {
   fill_mesh_and_initialize_test_fields("generated:2x2x2");
 
-  do_the_test(bulk);
+  test_ngp_mesh_1(bulk);
 }
 
+void test_ngp_mesh_field_values(const stk::mesh::BulkData& bulk, 
+                                VectorFieldType* velocity, 
+                                GenericFieldType* massFlowRate)
+{
+  const stk::mesh::MetaData& meta = bulk.mesh_meta_data();
+  stk::mesh::Selector all_local = meta.universal_part() & meta.locally_owned_part();
+  const stk::mesh::BucketVector& elemBuckets = bulk.get_buckets(stk::topology::ELEM_RANK, all_local);
+
+  ngp::Mesh ngpMesh(bulk);
+  ngp::Field<double> ngpVelocity(bulk, *velocity);
+  ngp::Field<double> ngpMassFlowRate(bulk, *massFlowRate);
+
+  const int bytes_per_team = 0;
+  const int bytes_per_thread = 0;
+  auto team_exec = sierra::nalu::get_device_team_policy(elemBuckets.size(), bytes_per_team, bytes_per_thread);
+
+  const double xVel = 1.0;
+  const double yVel = 2.0;
+  const double zVel = 3.0;
+  const double flowRate = 4.0;
+
+  Kokkos::parallel_for(team_exec, KOKKOS_LAMBDA(const sierra::nalu::DeviceTeamHandleType& team)
+  {
+    const ngp::Mesh::BucketType& b = ngpMesh.get_bucket(stk::topology::ELEM_RANK, team.league_rank());
+
+    const size_t bucketLen   = b.size();
+    const size_t simdBucketLen = sierra::nalu::get_num_simd_groups(bucketLen);
+ 
+    Kokkos::parallel_for(Kokkos::TeamThreadRange(team, simdBucketLen), [&](const size_t& bktIndex)
```

Comment:
So a question: when to use `KOKKOS_LAMBDA` (as you did for team `parallel_for`) vs. `[&](...)`?

Reply:
My understanding is that it's good to use KOKKOS_LAMBDA for the outer parallel_for. KOKKOS_LAMBDA captures using '=' which makes the captured things const. (Because things from the host can't be modified in the device kernel.) But the inner loop can use non-const capture because it's being launched from already on device. Maybe that breaks down in some cases but that's my understanding at this point.

Reply:
@alanw0 I didn't realize there was a `const ([=])` or `non-const ([&])` implication to these two. I thought `[=]` implied a copy (pass by value?) instead of pass by reference. Did you mean the call site essentially looks like a const because it is pass by value?

Reply:
@sayerhs I think you are precisely correct. I was thinking about constness with respect to the calling scope, but to be precise the difference is pass by value vs reference.